templates/expo-template-tabs/navigation/MainTabNavigator.js

```diff
@@ -1,79 +1,71 @@
 import React from 'react';
 import { Platform } from 'react-native';
-import { createStackNavigator } from 'react-navigation-stack';
-import { createBottomTabNavigator } from 'react-navigation-tabs';
-
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import { createStackNavigator } from '@react-navigation/stack';
 import TabBarIcon from '../components/TabBarIcon';
 import HomeScreen from '../screens/HomeScreen';
 import LinksScreen from '../screens/LinksScreen';
 import SettingsScreen from '../screens/SettingsScreen';
 
-const config = Platform.select({
-  web: { headerMode: 'screen' },
-  default: {},
-});
-
-const HomeStack = createStackNavigator(
-  {
-    Home: HomeScreen,
-  },
-  config
-);
-
-HomeStack.navigationOptions = {
-  tabBarLabel: 'Home',
-  tabBarIcon: ({ focused }) => (
-    <TabBarIcon
-      focused={focused}
-      name={
-        Platform.OS === 'ios'
-          ? `ios-information-circle${focused ? '' : '-outline'}`
-          : 'md-information-circle'
-      }
-    />
-  ),
-};
-
-HomeStack.path = '';
-
-const LinksStack = createStackNavigator(
-  {
-    Links: LinksScreen,
-  },
-  config
-);
-
-LinksStack.navigationOptions = {
-  tabBarLabel: 'Links',
-  tabBarIcon: ({ focused }) => (
-    <TabBarIcon focused={focused} name={Platform.OS === 'ios' ? 'ios-link' : 'md-link'} />
-  ),
-};
-
-LinksStack.path = '';
-
-const SettingsStack = createStackNavigator(
-  {
-    Settings: SettingsScreen,
-  },
-  config
-);
-
-SettingsStack.navigationOptions = {
-  tabBarLabel: 'Settings',
-  tabBarIcon: ({ focused }) => (
-    <TabBarIcon focused={focused} name={Platform.OS === 'ios' ? 'ios-options' : 'md-options'} />
-  ),
-};
-
-SettingsStack.path = '';
-
-const tabNavigator = createBottomTabNavigator({
-  HomeStack,
-  LinksStack,
-  SettingsStack,
-});
+const TabNavigator = createBottomTabNavigator();
 
-tabNavigator.path = '';
+const Stack = createStackNavigator();
 
-export default tabNavigator;
+export default function MainTabNavigator() {
```

Comment:
Maybe we should this nesting? Instead of stacks in tabs, it'll be better to have tabs in stack. The reason is that people often don't want to see the tab bar after pushing a new screen. With this structure, they need to hide the tab bar manually after pushing the screen which is extra code and can also screw up the animation since screen height changes during the animation.

Reply:
What do you think about a stack at the root, then a tab in the initial screen of that stack? It'd also simplify the deep linking config a bit.

Reply:
How I handle this is creating the Tabs inside a Stack along with other screen?

So when we navigate to a new screen, the Tabs hides seamlessly.

Reply:
I think that it's ok to have if there are no issues. I can wrap it in a factory, but I want to modify the header title, and according to the fact that this is an example app, I don't want to use `dangerouslyGetParent`

Reply:
> but I want to modify the header title
it doesn't need `dangerouslyGetParent`, but you can leave it out. this is a template, should be simple.